### PR TITLE
test: add table-driven tests for zero-coverage CLI command functions

### PR DESCRIPTION
**Pull request by Aureliolo**
## Summary

Adds table-driven tests for previously zero-coverage pure functions in the Go CLI, targeting the two highest-risk paths: data destruction (`stop --volumes`) and command injection (`logs` input validation). A small 64-bit-only cleanup and a doc-precision fix surfaced during review are folded in.

- **`buildDownArgs`** (`stop.go`): pins exactly when `--volumes` is emitted (the data-destruction flag), `--timeout` parsing (non-negative / whole-second bounds), and exact arg ordering.
- **`validateLogsInput`** (`logs.go`): pins the injection-safety gate. Accept + reject cases for `--tail`, flag-shaped `--since`/`--until`, and service names (space, `;`, `/`, `$()`, `!`, newline, null byte, backtick, pipe).
- **`buildLogsArgs`** (`logs.go`): full flag-to-args mapping, plus the load-bearing `--` separator assertion (a flag-shaped service like `--evil` must sit after `--`, never before).
- **`isImageInUse` / `emitCleanupSummary`** (`cleanup.go`): in-use vs hard-failure classification (incl. wrapped errors and the case-sensitivity boundary) and the freed / removed / skipped output branches.
- **`validateCleanupFlags`** (`flags_test.go`): boundary cases (`keep=0`, large keep) and `--all` independence from keep validation.
- **cleanup**: drop the `int64`-widening + 32-bit-build comment on the `docker_sock_gid` validators in `config/validate.go` and `compose/validate.go`. Build targets are 64-bit only (`goarch: [amd64, arm64]`), so the bare comparison compiles on every matrix cell.
- **docs**: clarify in `cli-commands.md` that `stop --timeout` defers to Docker Compose's 10s default (the CLI passes no `--timeout` unless set).

## Test plan

- `go -C cli test ./...` (all packages green)
- `gofmt -l cli/` (clean) and `go -C cli vet ./...` (clean)
- `golangci-lint run` (0 issues)

## Review coverage

Pre-PR review ran 5 agents (go-reviewer, go-conventions-enforcer, go-security-reviewer, docs-consistency, comment-quality-rot). go-reviewer approved; all valid findings addressed. No linked issue (isolated test-hardening).

### CODE REVIEW

File: cli/internal/compose/validate.go [hunk 1]
```diff
@@ -102,10 +102,9 @@ func validateSandbox(p Params) error {
 	if strings.ContainsAny(p.DockerSock, "\"'`$\n\r{}[]") {
 		return fmt.Errorf("docker socket path %q contains unsafe characters", p.DockerSock)
 	}
-	// p.DockerSockGID is an int; the upper bound 4294967295 (uint32 max)
-	// is not representable in a 32-bit signed int, so widen the
-	// comparison to int64 to keep the check correct on 32-bit builds.
-	if gid := int64(p.DockerSockGID); gid < -1 || gid > 4294967295 {
+	// docker_sock_gid is a Linux GID: -1 disables the override, the upper
+	// bound is uint32 max.
+	if p.DockerSockGID < -1 || p.DockerSockGID > 4294967295 {
```
[the remark this answers is not in the document]
Comment (Aureliolo) [reply cli/internal/compose/validate.go]: Restored the int64 widening in commit 2a7ca552 (both compose/validate.go and config/validate.go now widen the GID to int64 before the bounds comparison, so the uint32-max constant is representable on 32-bit toolchains). Thanks for the catch.

File: docs/reference/cli-commands.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ Defaults shown below are compiled into the binary; tunable defaults are also rea
 | `start` | `--dry-run` | `false` | Show what would happen without executing |
 | `start` | `--no-detach` | `false` | Run in foreground (stream logs; Ctrl+C to stop) |
 | `start` | `--no-verify` | `false` | Skip image signature verification (also `--skip-verify`) |
-| `stop` | `--timeout` / `-t` | `10s` | Graceful shutdown timeout |
+| `stop` | `--timeout` / `-t` | `10s` (Docker Compose default) | Graceful shutdown timeout (the CLI passes no `--timeout` unless set, deferring to Docker Compose) |
```
[the remark this answers is not in the document]
Comment (Aureliolo) [reply docs/reference/cli-commands.md:32]: Declining this one. docs/reference/cli-commands.md is not in check_doc_numeric_macros _SCOPED_FILES (README.md, docs/index.md, docs/roadmap/index.md, docs/architecture/decisions.md, docs/reference/convention-gates.md); the gate docstring explicitly puts references and per-command tables out of scope, and the matcher only flags numbers adjacent to stat nouns (tests/providers/agents/stars/releases). Also, 10s is Docker Compose own built-in default (the CLI passes no --timeout unless set), not a SynthOrg-generated runtime stat: data/runtime_stats.yaml is regenerated by generate_runtime_stats.py from code scans, so a hand-added key would fail check_runtime_stats_freshness. @coderabbitai resolve
